news/_posts/2020-09-30-how-to-write-a-new-ui-test.md

```diff
@@ -0,0 +1,386 @@
+---
+layout: post
+title:  "Automated UI tests tutorial"
+subtitle: "Step by step - Adding a new PrestaShop UI test"
+date:  2020-09-30 12:00:00
+authors: [boubkerbribri]
+icon: icon-github-alt
+image: /assets/images/theme/meta-logo-build.png
+tags: [community, contribution, tests, automation]
+---
+
+One year ago, the QA team started to develop a new test framework, based on [Puppeteer](https://github.com/puppeteer/puppeteer), [Mocha](https://mochajs.org/), and [Chai](https://www.chaijs.com/). You can read more about the framework in a previous [article](https://build.prestashop.com/news/the-new-prestashop-test-framework/).
+Since then, the tests coverage has been continually increasing and we recently improved our framework by switching to [Playwright](https://playwright.dev/).
+
+In this article, we will explain how to create a new UI (User Interface) test for the [PrestaShop core project](https://github.com/PrestaShop/PrestaShop).
+
+Before we start, you should definitely read about our stack and architecture in the [PrestaShop devdoc](https://devdocs.prestashop.com/1.7/testing/ui-tests/how-to-contribute-and-create-ui-tests/). 
+
+
+## I. Writing the scenario
+
+The first step to writing your scenario is to identify exactly what you want to test, and limit the test's scope to that. For this example **we will be checking that the customer link in the Back Office "Orders" page redirects to the "view customer" page**.
+The second step involves a manual check of this very scenario to be sure it's working, and to write down all the steps needed, from start to finish.
+
+Now that we have the scenario, we can create a new javascript file and write our scenario using Mocha (see example below).
+
+The directory in which we create our file should be chosen wisely (which campaign? BO or FO? Which page in BO? ...). This file organization is very important since Mocha lets you run test by folders (recursively or not), so grouping your tests by features or meta-features (pages) is a very good idea.
+
+```js
+// 'describe' = scenario
+// 'it' = step
+
+describe('View customer form orders page', async function(){
+  it('should login in BO');
+
+  it('should go to orders page');
+
+  it('should reset all filters');
+
+  it('should filter orders by customer name');
+
+  it('should check customer link');
+
+});
+```
+
+Note 1: 
```

Comment:
```suggestion

```

Reply:
Not fixed on my end ? Idk if it's normal

Reply:
didn't push that one, my bad, should be good now. But The article is still a WIP